File: client-clojure/src/sirqul_io_t_platform/specs/activity_response.clj
(ns sirqul-io-t-platform.specs.activity-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def activity-response-data
  {
   (ds/opt :activityId) int?
   (ds/opt :created) int?
   (ds/opt :appKey) string?
   (ds/opt :targetId) int?
   (ds/opt :targetType) string?
   (ds/opt :targetLegacyId) string?
   (ds/opt :targetDescription) string?
   (ds/opt :targetActive) boolean?
   (ds/opt :action) string?
   (ds/opt :slaveEntityId) int?
   })

(def activity-response-spec
  (ds/spec
    {:name ::activity-response
     :spec activity-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ActivityResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ActivityResponse._

case class ActivityResponse (
  activityId: Option[Long],
created: Option[Long],
appKey: Option[String],
targetId: Option[Long],
targetType: Option[String],
targetLegacyId: Option[String],
targetDescription: Option[String],
targetActive: Option[Boolean],
action: Option[String],
slaveEntityId: Option[Long])

object ActivityResponse {
  import DateTimeCodecs._

  implicit val ActivityResponseCodecJson: CodecJson[ActivityResponse] = CodecJson.derive[ActivityResponse]
  implicit val ActivityResponseDecoder: EntityDecoder[ActivityResponse] = jsonOf[ActivityResponse]
  implicit val ActivityResponseEncoder: EntityEncoder[ActivityResponse] = jsonEncoderOf[ActivityResponse]
}
